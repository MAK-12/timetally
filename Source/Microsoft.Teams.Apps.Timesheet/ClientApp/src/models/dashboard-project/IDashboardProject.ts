// <copyright file="IDashboardProject.ts" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

import { Guid } from "guid-typescript";

export interface IDashboardProject {
    id: Guid;
    title: string;
    utilizedHours: number;
    totalHours: number;
}