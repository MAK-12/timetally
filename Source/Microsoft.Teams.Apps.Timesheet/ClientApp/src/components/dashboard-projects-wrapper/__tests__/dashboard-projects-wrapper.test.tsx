// <copyright file="create-event-wrapper.test.tsx" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

import * as React from "react";
import DashboardProjectWrapper from "../dashboard-projects-wrapper";
import { Provider } from "@fluentui/react-northstar";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import { IDashboardProject } from "../../../models/dashboard/dashboard-project";
import { Guid } from "guid-typescript";

jest.mock("react-i18next", () => ({
    useTranslation: () => ({
        t: (key: any) => key,
        i18n: { changeLanguage: jest.fn() },
    }),

    withTranslation: () => (Component: any) => {
        Component.defaultProps = {
            ...Component.defaultProps,
            t: (key: any) => key,
        };
        return Component;
    },
}));
jest.mock("@microsoft/teams-js", () => ({
    initialize: () => {
        return true;
    },
    getContext: (callback: any) =>
        callback(
            Promise.resolve({ teamId: "ewe", entityId: "sdsd", locale: "en-US" })
        ),
}));

let container: any = null;
let projects: Array<IDashboardProject> = [
    { id: Guid.create(), title: "Project 1", totalHours: 22, utilizedHours: 10 },
    { id: Guid.create(), title: "Project 2", totalHours: 100, utilizedHours: 0 },
];
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    // container *must* be attached to document so events work correctly.
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

describe("DashboardProjectWrapper", () => {
    it("renders snapshots", () => {
        act(() => {
            render(
                <Provider>
                    <DashboardProjectWrapper isMobileView={false} projects={projects} onProjectCardClick={() => { }} />
                </Provider>,
                container
            );
        });

        expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
});