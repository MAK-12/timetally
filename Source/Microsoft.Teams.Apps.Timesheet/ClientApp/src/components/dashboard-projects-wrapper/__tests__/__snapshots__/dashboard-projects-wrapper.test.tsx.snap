// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardProjectWrapper renders snapshots 1`] = `
"<div class=\\"ui-provider\\" dir=\\"ltr\\" data-uses-unhanded-props=\\"true\\">
  <div class=\\"ui-flex\\" data-uses-unhanded-props=\\"true\\">
    <div>
      <div class=\\"ui-flex project-card-container\\" data-uses-unhanded-props=\\"true\\"><svg class=\\"CircularProgressbar circular-progress\\" viewBox=\\"0 0 100 100\\" data-test-id=\\"CircularProgressbar\\">
          <path class=\\"CircularProgressbar-trail\\" style=\\"stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 0px;\\" d=\\"
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    \\" stroke-width=\\"8\\" fill-opacity=\\"0\\"></path>
          <path class=\\"CircularProgressbar-path\\" style=\\"stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 156.07432303034093px;\\" d=\\"
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    \\" stroke-width=\\"8\\" fill-opacity=\\"0\\"></path><text class=\\"CircularProgressbar-text\\" x=\\"50\\" y=\\"50\\">46%</text>
        </svg>
        <div class=\\"ui-flex text-container\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text project-title\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">Project 1</span><span class=\\"ui-text project-subtitle\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">10/22 hoursUtilizeLabel</span><span class=\\"ui-text underutilized-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">underutilizedLabel</span></div>
      </div>
    </div>
    <div>
      <div class=\\"ui-flex project-card-container\\" data-uses-unhanded-props=\\"true\\"><svg class=\\"CircularProgressbar circular-progress\\" viewBox=\\"0 0 100 100\\" data-test-id=\\"CircularProgressbar\\">
          <path class=\\"CircularProgressbar-trail\\" style=\\"stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 0px;\\" d=\\"
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    \\" stroke-width=\\"8\\" fill-opacity=\\"0\\"></path>
          <path class=\\"CircularProgressbar-path\\" style=\\"stroke-dasharray: 289.02652413026095px 289.02652413026095px; stroke-dashoffset: 289.02652413026095px;\\" d=\\"
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    \\" stroke-width=\\"8\\" fill-opacity=\\"0\\"></path><text class=\\"CircularProgressbar-text\\" x=\\"50\\" y=\\"50\\">0%</text>
        </svg>
        <div class=\\"ui-flex text-container\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text project-title\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">Project 2</span><span class=\\"ui-text project-subtitle\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">0/100 hoursUtilizeLabel</span><span class=\\"ui-text underutilized-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">underutilizedLabel</span></div>
      </div>
    </div>
  </div>
</div>"
`;
